Develop

SergeyZh89 opened the pull request.

SergeyZh89 committed:
feat: Удаление фильмов и пользователей

SergeyZh89 committed:
feat: Удаление фильмов и пользователей v2

SergeyZh89 committed:
feat: Удаление фильмов и пользователей v3

SergeyZh89 committed:
Merge pull request #7 from SergeyZh89/add-remove-endpoint

Add remove endpoint

NekhoroshkinaElena committed:
feat: добавить функциональность "Общие фильмы"

Commit:
Init add-recommendations

Commit:
build test

Commit:
build test

SergeyZh89 committed:
Merge pull request #9 from SergeyZh89/add-common-films

Add common films

Commit:
add changes

SergeyZh89 committed:
feat: director SERVICE, DAO, DAOIMPL, CONTROLLER

SergeyZh89 committed:
Merge pull request #11 from SergeyZh89/add-controller

feat: director SERVICE, DAO, DAOIMPL, CONTROLLER

NekhoroshkinaElena committed:
feat: добавить функциональность "Общие фильмы"

NekhoroshkinaElena committed:
feat: добавить таблицы director, film_director, создать новый класс Director, добавить новое поле в класс Film

KiselevS committed:
add most popular

SergeyZh89 committed:
Merge pull request #12 from SergeyZh89/add-most-populars

Add most populars

NekhoroshkinaElena committed:
feat: добавить таблицы director, film_director, создать новый класс Director, добавить новое поле в класс Film

NekhoroshkinaElena committed:
feat: добавить таблицы director, film_director, создать новый класс Director, добавить новое поле в класс Film

SergeyZh89 committed:
Merge pull request #13 from SergeyZh89/add-table-director

Add table director

NekhoroshkinaElena committed:
Merge remote-tracking branch 'origin/develop' into add-director

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/dao/impl/FilmDaoImpl.java

NekhoroshkinaElena committed:
feat: добавить метод для получения фильмов режиссера, отсортированных по количеству лайков или году выпуска

SergeyZh89 committed:
feat: Поиск

SergeyZh89 committed:
Merge pull request #14 from SergeyZh89/add-director

Add director

SergeyZh89 committed:
Merge remote-tracking branch 'origin/develop' into add-search

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/controllers/FilmController.java
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java

Commit:
add-recommendations

Commit:
add-recommendations resolve conflicts

SergeyZh89 committed:
feat: Поиск2

SergeyZh89 committed:
Merge pull request #15 from SergeyZh89/add-recommendations

Add recommendations

SergeyZh89 committed:
Merge remote-tracking branch 'origin/develop' into add-search

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/dao/FilmDao.java
#	src/main/java/ru/yandex/practicum/filmorate/service/UserService.java

SergeyZh89 committed:
feat: Поиск complete

SergeyZh89 committed:
Merge pull request #16 from SergeyZh89/add-search

Add search

goaliepash committed:
feat: создание отзывов

goaliepash committed:
bugfix: изменение метода updateReview

SergeyZh89 committed:
Merge pull request #17 from SergeyZh89/add-reviews

Функциональность "Отзывы"

NekhoroshkinaElena committed:
feat: добавить новую функциональность "лента событий"

NekhoroshkinaElena committed:
Merge branch 'add-feed' into develop

NekhoroshkinaElena committed:
feat: добавить новую функциональность "лента событий"

SergeyZh89 committed:
Merge pull request #18 from SergeyZh89/add-feed

Add feed

NekhoroshkinaElena committed:
fix: исправить ошибку при добавлении лайка фильму

SergeyZh89 committed:
Merge pull request #20 from SergeyZh89/add-feed

fix: исправить ошибку при добавлении лайка фильму

utausheva commented:
Сергей, привет) Добавьте, пожалуйста, выполненные задания сюда (в описание/комментарий) или в ридми файл (это обязательно). А еще вдруг у вас есть обновленная версия схемы базы данных, тоже полезный файл (это желательно, если есть)

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/dao/impl/DirectorDaoImpl.java:
```diff
@@ -0,0 +1,87 @@
+                "d.name " +
+                "FROM director AS d " +
+                "WHERE d.id = ?;";
+        return Optional.ofNullable(jdbcTemplate.query(sqlQuery, (rs, rowNum) -> makeDirector(rs), id)
```
Если объекта нет в базе, то бросается исключение.. зачем тогда возвращать Optional? Расскажите о вашей идее такого подхода)

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/dao/impl/DirectorDaoImpl.java:
```diff
@@ -0,0 +1,87 @@
+
+    @Override
+    public Director updateDirector(Director director) {
+        if (getDirectorById(director.getId()).isEmpty()) {
```
а это не задача сервиса - провалидировать входящие данные?

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/dao/impl/RatingMpaDaoImpl.java:
```diff
@@ -13,7 +13,7 @@
 @Slf4j
```
логгер здесь точно нужен? (проверьте остальные классы тоже)

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/dao/impl/RecommendationsDaoImpl.java:
```diff
@@ -0,0 +1,68 @@
+
+    @Override
+    public List<Film> getRecommendationsByUser(long userId, int recCount) {
+        String sql = "WITH ul AS (\n" +
```
🔥🔥

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/dao/impl/ReviewDaoImpl.java:
```diff
@@ -0,0 +1,133 @@
+
+    @Override
+    public void like(long id, long userId) {
+        jdbcTemplate.update("DELETE FROM review_dislikes WHERE review_id = ? AND user_id = ?;", id, userId);
```
а надо удалять дизлайк при добавлении лайка?)

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/dao/impl/ReviewDaoImpl.java:
```diff
@@ -0,0 +1,133 @@
+    }
+
+    @Override
+    public void like(long id, long userId) {
```
like в данном контексте является существительным, методы принято называть глаголами, давайте переименуем в addLike

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/dao/impl/ReviewDaoImpl.java:
```diff
@@ -0,0 +1,133 @@
+
+    @Override
+    public void dislike(long id, long userId) {
+        jdbcTemplate.update("DELETE FROM review_likes WHERE review_id = ? AND user_id = ?;", id, userId);
```
тоже: удалять дизлайк при добавлении лайка не нужно

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/dao/impl/UserDaoImpl.java:
```diff
@@ -20,9 +23,11 @@
 @Component
 public class UserDaoImpl implements UserDao {
     private final JdbcTemplate jdbcTemplate;
+    FeedService feedService;
```
Дао не должно зависеть ни от других дао, ни от любых сервисов тем более. Нужно пользоваться прямыми запросами в базу.

utausheva replied:
а любую бизнес-логику нужно реализовывать в сервисах

SergeyZh89 committed:
Update README.md

Добавлены:
Функциональность "Поиск",
Функциональность "Лента событий",
Функциональность "Отзывы",
Функциональность "Удаление фильмов и пользователей",
Функциональность "Общие фильмы",
Функциональность "Рекомендации",
Функциональность "Добавление режиссеров в фильмы",
Функциональность "Вывод самых популярных фильмов по жанру и годам".

SergeyZh89 commented:
Добавлены:
Функциональность "Поиск",
Функциональность "Лента событий",
Функциональность "Отзывы",
Функциональность "Удаление фильмов и пользователей",
Функциональность "Общие фильмы",
Функциональность "Рекомендации",
Функциональность "Добавление режиссеров в фильмы",
Функциональность "Вывод самых популярных фильмов по жанру и годам".

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/dao/impl/UserDaoImpl.java:
```diff
@@ -34,6 +39,7 @@ public List<User> getUsers() {
     @Override
     public List<User> getFriends(long id) {
         String sql = "SELECT * FROM USERS WHERE ID IN (SELECT FRIENDS_ID FROM USER_FRIENDS WHERE USER_ID=?)";
+        getUser(id);
```
это валидация (проверка на существование пользователя), лучше ее сделать ДО запроса в бд и в сервисе

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/exceptions/GlobalExceptionHandler.java:
```diff
@@ -9,14 +9,13 @@
     }
 
-    @ExceptionHandler({UserNotFoundException.class, FilmNotFoundException.class, GenreNotFoundException.class, RatindMpaNotFoundException.class})
+    @ExceptionHandler({UserNotFoundException.class, FilmNotFoundException.class, GenreNotFoundException.class, RatindMpaNotFoundException.class, DirectorNotFoundException.class, ReviewNotFoundException.class})
```
слишком длинная строка

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/exceptions/UserNotFoundException.java:
```diff
@@ -1,7 +1,6 @@
 package ru.yandex.practicum.filmorate.exceptions;
 
 public class UserNotFoundException extends RuntimeException {
-
```
а в классе ReviewNotFoundException есть пустая строка между названием класса и первым методом) Давайте сделаем везде одинаково (на мой взгляд лучше с пустой строкой, но это на выбор команды)

utausheva commented on src/main/java/ru/yandex/practicum/filmorate/service/ReviewService.java:
```diff
@@ -0,0 +1,115 @@
+@Service
+public class ReviewService {
+
+    private final ReviewDaoImpl reviewDao;
```
Объявлять нужно не классы, а интерфейсы. Полиморфизм) Поправьте во всех классах сервисов

utausheva reviewed (commented):
Привет всем!
Вы проделали огромную работу в команде, очень здорово, что получилось (надеюсь) сработаться)))
Отлично получились сложные запросы в базу, мне очень понравилось, что отрефакторили старый код (заметила, что некоторых важных ошибок не увидела на предыдущих спринтах в этой работе), значит каждый работал не только над своим функционалом, но и "болел" за продукт в целом, это круто! Сделали даже тесты, хотя это не требовалось по тз) Вы постарались подстроить свой стиль под командный (кое-где заметно, что писали несколько человек, но в основном всё методы однообразны, это круто!). Отдельно отмечу отличную работу тимлида команды, очень оперативно добавлялись данные по запросу в слаке ;)
Модель получилась без нареканий, спасибо за перечень реализованных фич! 
Теперь про ревью: основное замечание-предложение в выполнении всей бизнес-логики в сервисах, а не в хранилищах, хранилища должны только добавлять/удалять/обновлять/возвращать данные. Остальные комментарии увидите в коде))
Замечания не сложные, должны справиться быстро и отправиться на заслуженные каникулы! Удачи)

goaliepash committed:
refactor: исправление замечаний

SergeyZh89 committed:
Merge pull request #22 from SergeyZh89/add-reviews

Исправление замечаний

utausheva reviewed (approved):
Всё круто!
Можно мерджить))
Вы супер! ☀️
